core/src/main/java/org/apache/calcite/rel/rules/JoinToCorrelateRule.java

```diff
@@ -35,6 +37,8 @@
 import org.apache.calcite.util.ImmutableBitSet;
 import org.apache.calcite.util.Util;
 
+import java.util.function.Function;
+
 /**
  * Rule that converts a {@link org.apache.calcite.rel.logical.LogicalJoin}
```

Comment:
I think the javadoc here should not explicitly refer to LogicalJoin and LogicalCorrelate. This part should be moved directly above the INSTANCE field.